### PR TITLE
Add binlog documentation

### LINKED ISSUE #2893: Add content to the developer guide documentation on how to collect binlogs
We should add this information to the docs 
https://github.com/microsoft/microsoft-ui-xaml/issues/2859#issuecomment-657115412

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Adds the steps explained by @stevenbrix in [this comment](https://github.com/microsoft/microsoft-ui-xaml/issues/2859#issuecomment-657115412) to the repo for easier reference.
## Motivation and Context

Closes #2893

### CODE REVIEW

File: docs/debugging_buildfailures.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Capturing binlogs
+Capturing and providing binlog files can help with debugging build and packaging issues. In order to collect binlogs, please follow these steps:
+
+1. Download the **[VS Project System Tools extension](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools)**
+2. Set the **Build Log File** verbosity to `Diagnostics`: `Tools->Options->Projects and Solutions->MSBuild project build log file verbosity`:<br/>
+![Screenshot of Build and run options](./images/binlogs_buildandrunoptions.png)
+
+3. Go to View->Other Windows->Build Logging:<br/>
+![Screenshot of Build Logging menu item](./images/binlogs_buildlogging_menuitem.png)
```
Comment (Felix-Dev) [inline docs/debugging_buildfailures.md:9]: As more and more images get created I think we should start creating subfolders for the image directory and thus sort images that way for easier maintenance and organization.

Comment (chingucoding) [reply docs/debugging_buildfailures.md]: Great idea! Moved them into a binlog folder for now.

**Comment by stevenbrix** [comment]
Thanks @chingucoding for sending this out! 

Do you think you could add some information on how to capture binlogs from the command line as well as visual studio? Those binlogs are generally more helpful and easier to diagnose.

Here's a helpful command for capturing a binlog that simulates the appx package upload to the store if you're having problems with Publish->Create App Packages

msbuild /p:AppxBundlePlatforms=x86 /p:Platform=x86 /p:Configuration=Release /p:BuildAppxUploadPackageForUap=true /bl

**Comment by chingucoding** [comment]
I've added a few words regarding the CLI, but I have to admit that I sort of just copied your steps from your comment into this file. I'm not an expert on this, and you definitely have more knowledge here than me.

**Pull request merged**